inductiva/_cli/cmd_tasks/list.py

```diff
@@ -16,14 +16,28 @@ def list_tasks(args, fout: TextIO = sys.stdout):
     or a single task with a specific ID
     or all tasks of a project.
     """
-    if args.project_name is not None:
-        print(f"Showing tasks for project: {args.project_name}.")
-        task_list = projects.Project(args.project_name).get_tasks()
-    elif args.task_id is not None:
-        task_list = [tasks.Task(args.task_id)]
+    project_name = args.project_name
+    task_id = args.task_id
+    last_n = args.last_n
+
+    if task_id is not None and (last_n is not None or project_name is not None):
+        print(
+            "inductiva tasks list: error: "
+            "argument id not allowed with argument --project-name or --last-n",
+            file=sys.stderr)
+        return 1
+
+    if project_name is not None:
+        print(f"Showing tasks for project: {project_name}.")
+        # With project the default last_n is -1 (all tasks)
+        last_n = -1 if args.last_n is None else args.last_n
+        task_list = projects.Project(project_name).get_tasks(last_n=last_n)
+    elif task_id is not None:
+        task_list = [tasks.Task(task_id)]
     else:
-        last_n = 5 if args.last_n is None else args.last_n
-        task_list = tasks.get(last_n=last_n)
+        # With no project the default last_n is 10
+        last_n = 10 if args.last_n is None else args.last_n
```

Comment:
This is a tricky thing to decide. If you specify a project it makes sense to list all the tasks for that project, but if you don't specify a project listing all tasks would not make sense. This is why we make the default value to 10.

I understand this can be confusing, I'm open to suggestions.

Reply:
I've made some changes.
We now have -a/--all options.
--all --last-n --id are mutually exclusive.
--id and --project-name are also mutually exclusive.
--last-n has the default value of 10 (when using projects or not).